typed-racket-lib/typed-racket/base-env/base-special-env.rkt

```diff
@@ -133,53 +133,33 @@
           (cl-> [((-HT a b)) (-seq (-pair a b))]
                 [(-HashTop) (-seq (-pair Univ Univ))]))]
   [(make-template-identifier 'default-in-immutable-hash 'racket/private/for)
-   (-poly (a b)
-          (cl-> [((-HT a b)) (-seq a b)]
-                [(-HashTop) (-seq Univ Univ)]))]
+   (-poly (a b) (-> (-Immutable-HT a b) (-seq a b)))]
```

Comment:
I don't think there is a backwards here. On Racket 6.5, this program gives me errors about `unsafe-hash-iterate-...` identifiers.

```
#lang typed/racket/base
(for ([(x y) (in-immutable-hash #hash((a . 1)))])
  (void))
```